ExtraHop/extrahop/reveal_360_trigger.py

```diff
@@ -0,0 +1,156 @@
+    frequency: int = 60
+    chunk_size: int = 1000
+
+
+class ExtraHopReveal360Connector(Connector):
+    module: ExtraHopModule
+    configuration: ExtraHopReveal360Configuration
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.context = PersistentJSON("context.json", self._data_path)
+        self.from_date = self.get_last_timestamp()
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(
+            auth=ExtraHopApiAuthentication(
+                base_url=self.module.configuration.base_url,
+                client_id=self.module.configuration.client_id,
+                client_secret=self.module.configuration.client_secret,
+            )
+        )
+
+    def get_last_timestamp(self) -> int:
+        now = int(time.time() * 1000)  # in milliseconds
+
+        with self.context as cache:
+            most_recent_ts_seen = cache.get("last_timestamp")
+
+        # if undefined, retrieve events from the last 7 days
+        if most_recent_ts_seen is None:
+            return now - 7 * 24 * 60 * 60 * 1000
+
+        one_week_ago = now - 7 * 24 * 60 * 60 * 1000
+        if most_recent_ts_seen < one_week_ago:
+            most_recent_ts_seen = one_week_ago
+
+        return most_recent_ts_seen
+
+    def set_last_timestamp(self, last_timestamp: int) -> None:
+        with self.context as cache:
+            cache["last_timestamp"] = last_timestamp
+
+    def fetch_page(self, from_time: int, offset: int) -> list[dict] | None:
+        params = {
+            "from": from_time,
+            "limit": self.configuration.chunk_size,
+            "offset": offset,
+            "sort": [{"direction": "asc", "field": "start_time"}],
+        }
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+        }
+        url = f"{self.module.configuration.base_url}/api/v1/detections/search"
+        response = self.client.post(url, json=params, headers=headers, timeout=60)
+
+        if not response.ok:
+            self.log(
+                message=(
+                    f"Request on ExtraHop Reveal(x) 360 API to fetch {response.url} "
+                    f"failed with status {response.status_code} - {response.reason}"
+                ),
+                level="error",
+            )
+            return None
+
+        content = response.json() if len(response.content) > 0 else None
+        return content
+
+    def iterate_through_pages(self, from_time: int) -> Generator[list, None, None]:
+        offset = 0
+
+        while self.running:
+            events = self.fetch_page(from_time=from_time, offset=offset)
+            if not events:
+                self.log(
+                    message=f"No events to forward. Waiting {self.configuration.frequency} seconds",
+                    level="info",
+                )
+                time.sleep(self.configuration.frequency)
```

Comment:
Do you think we can pause the connector in the `next_batch` method? I fear pausing the connector here, after fetching the last page, will prevent us to catch back in we have some delay.